unittests/api_tests.cpp

```diff
@@ -929,12 +947,62 @@ BOOST_AUTO_TEST_CASE(checktime_fail_tests) { try {
       time_left_in_block_us -= increment;
    }
    BOOST_CHECK_EXCEPTION( call_test( t, test_api_action<TEST_METHOD("test_checktime", "checktime_failure")>{},
-                                    0, 200, fc::raw::pack(10000000000000000000ULL) ),
+                                    0, 200, 200, fc::raw::pack(10000000000000000000ULL) ),
                           block_cpu_usage_exceeded, is_block_cpu_usage_exceeded );
 
    BOOST_REQUIRE_EQUAL( t.validate(), true );
 } FC_LOG_AND_RETHROW() }
 
+BOOST_AUTO_TEST_CASE(checktime_pause_test) { try {
+
+   fc::temp_directory tempdir;
+   auto conf_genesis = tester::default_config( tempdir );
+
+   auto& cfg = conf_genesis.second.initial_configuration;
+
+   cfg.max_block_cpu_usage        = 150'000;
+   cfg.max_transaction_cpu_usage  = 24'999; // needs to be large enough for create_account and set_code
+   cfg.min_transaction_cpu_usage  = 1;
+
+   tester t( conf_genesis.first, conf_genesis.second );
+   t.execute_setup_policy( setup_policy::full );
+
+   t.produce_blocks(2);
+
+   ilog( "create account" );
+   t.create_account( "pause"_n );
+   ilog( "set code" );
+   t.set_code( "pause"_n, contracts::test_api_wasm() ); // only test that should use pause account so it is not cached
+   ilog( "produce block" );
+   t.produce_blocks(1);
+
+   int64_t ram_bytes; int64_t net; int64_t cpu;
+   auto& rl = t.control->get_resource_limits_manager();
+   rl.get_account_limits( "pause"_n, ram_bytes, net, cpu );
+   BOOST_CHECK_EQUAL( cpu, -1 );
+   auto cpu_limit = rl.get_block_cpu_limit();
+   idump(("cpu_limit")(cpu_limit));
+   BOOST_CHECK( cpu_limit <= 150'000 );
+
+   // Verify that the restriction on the transaction of 24'999 is honored even though there is wall clock time to
+   // load the wasm. If this test fails it is possible that the wasm loaded faster or slower than expected.
```

Comment:
I think we need two tests
- The 24999 limit *is* extended by a pause (this check)
- The block deadline *is not* extended by a pause (I don't see this)

Reply:
Oh, and one more:
- The cpu limit is is originally before the block deadline, but is extended past it by a pause.

Reply:
Added tests.